### ISSUE TITLE
[2.7.0] Build failed for Mac Catalyst, Xcode 11.4

### ISSUE THREAD

**Issue by AlexG-MK** [mixpanel/mixpanel-swift#355]
App doesn't build for Mac Catalyst, XCode 11.4 target with errors:

mixpanel-swift-2.7.0/Mixpanel/MixpanelInstance.swift:258:32: Use of unresolved identifier 'CTTelephonyNetworkInfo'

mixpanel-swift-2.7.0/Mixpanel/MixpanelInstance.swift:392:51: Type 'NSNotification.Name?' has no member 'CTRadioAccessTechnologyDidChange'

mixpanel-swift-2.7.0/Mixpanel/PushNotifications.swift:145:27: 'shared' is unavailable in application extensions for Mac Catalyst: Use view controller based solutions where appropriate instead.

mixpanel-swift-2.7.0/Mixpanel/PushNotifications.swift:145:34: 'open(_:options:completionHandler:)' is unavailable in application extensions for Mac Catalyst

**Comment by AlexG-MK** [comment]
App doesn't build for Mac Catalyst, XCode 11.4 target with errors:

mixpanel-swift-2.7.0/Mixpanel/MixpanelInstance.swift:258:32: Use of unresolved identifier 'CTTelephonyNetworkInfo'

mixpanel-swift-2.7.0/Mixpanel/MixpanelInstance.swift:392:51: Type 'NSNotification.Name?' has no member 'CTRadioAccessTechnologyDidChange'


mixpanel-swift-2.7.0/Mixpanel/PushNotifications.swift:145:27: 'shared' is unavailable in application extensions for Mac Catalyst: Use view controller based solutions where appropriate instead.

mixpanel-swift-2.7.0/Mixpanel/PushNotifications.swift:145:34: 'open(_:options:completionHandler:)' is unavailable in application extensions for Mac Catalyst

**Comment by jlaws** [comment]
+1

**Comment by ph1lb4** [comment]
Any progress on this? We cannot ship a new build of our product unless this is fixed. Even going back a few releases, it shows the same error (e.g. 2.6.8).

**Comment by mman** [comment]
I'm on 2.6.6 with DECIDE turned OFF compiling for Catalyst just fine. Not that it fixes the issue above, but just to confirm that you can go to production if needed.

**Comment by ph1lb4** [comment]
Thanks for letting me know. Is there any documentation on turning "DECIDE" to "OFF", I am not aware about that setting and how to modify it?

**Comment by mman** [comment]
@ph1lb4 It's a swift compilation flag turned by default to ON in .xcodeproj (https://github.com/mixpanel/mixpanel-swift/blob/master/Mixpanel.xcodeproj/project.pbxproj#L1287) and in podspec (https://github.com/mixpanel/mixpanel-swift/blob/master/Mixpanel.xcodeproj/project.pbxproj#L1287) and is only useful when using mixpanel in development, for A/B testing etc. As far as I understand the code anyway. If you turn it off, you leave out a lot of problematic code that needs updates and contained crashes in the past etc.

Unfortunately there is no easy way to override the setting IMHO so I had to fork and remove it for my own use. Would be a good PR to make this configurable in an easier way.

**Comment by andygeers** [comment]
@mman Do you have such a fork available for others to use? Or does it contain other changes as well?

**Comment by mman** [comment]
I have it here https://github.com/mman/mixpanel-swift but no guarantees it will be kept stable, it's a private fork for just me and I use it by including the .xcodeproj into my own projects. No support offered :)

**Comment by zihejia** [comment]
Sorry guys, we will fix this shortly.

**Comment by gregorpichler** [comment]
@zihejia any news on this? We can't ship an app update atm :/

**Comment by ph1lb4** [comment]
I have forked this here into a buildable version for Catalyst: https://github.com/Butleroy/mixpanel-swift

Please note that there is absolutely no guarantee that I have not bricked substantial features and I advise you only use this for dev purposes, until the real fix comes along.

**Comment by emreoktem** [comment]
> Sorry guys, we will fix this shortly.

@zihejia Do you have date for official fix?

**Comment by zihejia** [comment]
hi @emreoktem , the fix has been merged into master. We will release a new version by tomorrow

**Comment by emreoktem** [comment]
@zihejia This is fantastic. Hope to see an update on cocoapods soon. Thanks.

**Comment by zihejia** [comment]
v2.7.1 has released, sorry guys for the delay. 
https://github.com/mixpanel/mixpanel-swift/releases/tag/v2.7.1. I'm closing this one now, please feel free to reopen it if the problem still exists.
